driver-core/src/main/com/mongodb/internal/connection/SocketStream.java

```diff
@@ -79,7 +82,9 @@ public void open() {
             inputStream = socket.getInputStream();
         } catch (IOException e) {
             close();
-            throw new MongoSocketOpenException("Exception opening socket", getAddress(), e);
```

Comment:
I think it's worth a ticket to make this consistent with read/write methods to throw IOException and do the exception transation in InternalStreamConnection.  It can be dependent on the ticket to remove Stream from the API.

Reply:
Created https://jira.mongodb.org/browse/JAVA-5179.